terraform/node_output.go

```diff
@@ -34,24 +37,73 @@ var (
 func (n *nodeExpandOutput) expandsInstances() {}
 
 func (n *nodeExpandOutput) temporaryValue() bool {
-	// this must always be evaluated if it is a root module output
+	// non root outputs are temporary
 	return !n.Module.IsRoot()
 }
 
 func (n *nodeExpandOutput) DynamicExpand(ctx EvalContext) (*Graph, error) {
+	if n.Destroy {
+		// if we're planning a destroy, we only need to handle the root outputs.
+		// The destroy plan doesn't evaluate any other config, so we can skip
+		// the rest of the outputs.
+		return n.planDestroyRootOutput(ctx)
+	}
+
+	var modules []addrs.ModuleInstance
+	switch {
+	case n.Module.IsRoot():
+		// root module don't expand
+		modules = append(modules, addrs.RootModuleInstance)
```

Comment:
Doesn't `expander.ExpandModule` already take care of this case, by returning just the root module instance if given the root module? :thinking:

Reply:
Thanks. This little snippet is left over from refactoring, and the switch is no longer even relevant since I ended up removing the use of this node entirely for root modules. I'll remove the switch from here before merging.